phases/ephemeral/witx/wasi_ephemeral_poll0.witx

```diff
@@ -0,0 +1,23 @@
+;; WASI Poll. This supports polling for events.
+;;
+;; Some content here is derived from [CloudABI](https://github.com/NuxiNL/cloudabi).
+
+(use "typenames.witx")
+
+(module $wasi_ephemeral_poll0
+  ;; Linear memory to be accessed by WASI functions that need it.
+  (import "memory" (memory))
+
+  ;; Concurrently poll for the occurrence of a set of events.
+  (@interface func (export "oneoff")
+    ;; The events to which to subscribe.
+    (param $in (@witx const_pointer $subscription_t))
```

Comment:
How and from what can a `subscription_t` instantiated? I would have expected an `fd_t` here, about which I would then ask how we exclude support for sockets and include support for files being polled.

Reply:
The type is described here:https://github.com/webassembly/wasi/blob/9df365b545fcf6fa12a8cb6dc47a23db76bba091/phases/ephemeral/witx/typenames.witx#L589
It is a tagged union that supports either a timer, a readable fd, or a writable fd.

Reply:
That's helpful, thanks @pchickey!